src/serac/physics/heat_transfer.hpp

```diff
@@ -625,6 +625,34 @@ class HeatTransfer<order, dim, Parameters<parameter_space...>, std::integer_sequ
     }
   }
 
+  /**
+   * @brief register a custom boundary integral calculation as part of the residual
+   *
+   * @tparam active_parameters a list of indices, describing which parameters to pass to the q-function
+   * @param qfunction a callable that returns the traction on a boundary surface
+   * @param optional_domain The domain over which the integral is computed
+   *
+   * ~~~ {.cpp}
+   *
+   *  heat_transfer.addCustomBoundaryIntegral(DependsOn<>{}, [](double t, auto position, auto temperature, auto
+   * temperature_rate){ auto [T, dT_dxi] = temperature; auto q           = 5.0*(T-25.0); return q;  // define a
+   * temperature-proportional heat-flux
+   *  });
+   *
+   * ~~~
+   *
+   * @note This method must be called prior to completeSetup()
+   */
+  template <int... active_parameters, typename callable>
+  void addCustomBoundaryIntegral(DependsOn<active_parameters...>, callable qfunction,
+                                 const std::optional<Domain>& optional_domain = std::nullopt)
+  {
+    Domain domain = (optional_domain.has_value()) ? optional_domain.value() : EntireBoundary(mesh_);
```

Comment:
just a style thing, since this construction appears in several places now
- `std::optional` can implicitly convert to a `bool` (equivalent to `has_value()`)
- you can also use the operator* to dereference it, rather than the `.value()`

so another way to write this is 

```cpp
Domain domain = (optional_domain) ? *optional_domain : EntireBoundary(mesh_);
```

Reply:
good to know, I like the brevity of your suggestion